bzt/modules/python/generators.py

```diff
@@ -823,9 +823,33 @@ def gen_module(self):
         stmts.append(self.gen_empty_line_stmt())
         stmts.extend(self.gen_global_vars())
         stmts.append(self.gen_empty_line_stmt())
+        stmts.extend(self.get_data_source_readers())
         stmts.append(self.gen_classdef())
         return ast.Module(body=stmts)
 
+
+    def get_data_source_readers(self):
+        readers = []
+        for idx, source in enumerate(self.scenario.get_data_sources(), start=1):
+            reader_tpl = "reader{idx} = CSVReaderPerThread({params})"
+
+            params = source.get("path")
+            if "loop" in source:
+                params += ", " + "loop=%s" % source.get("loop")
+
+            if "quoted" in source:
+                params += ", " + "quoted=%s" % source.get("quoted")
+
+            if "delimiter" in source:
+                params += ", " + "delimiter='%s'" % source.get("delimiter")
```

Comment:
`%r` is better than `%s` when generating python code like this. It deals with escaping.